### ISSUE TITLE
Option to select 13-sector Disk II ROM

### ISSUE THREAD

**Issue by skiphipdata** [AppleWin/AppleWin#1133]
It would be nice to have a command line switch to set the Disk II to 13-sector support, so as to not have to use a boot utility. Or, what might be even nicer would an option to use a custom Disk II ROM image, similar to the option to select a custom main ROM image--since the only difference between a 16-sector and 13-sector controller is the ROM that is in use.

**Comment by tomcw** [comment]
If you are using a .woz image, then it contains metadata about whether the boot track is 13-sector or not.
If it's 13-sector, then AppleWin will automatically switch the Disk II controller card's firmware to be the 13-sector ROM.
(And it'll switch back on inserting a 16-sector woz image _after a reset_.)

From the first comment:
>NB. No plans to support the rare .d13 format. And no plans to auto-detect whether a .nib is 13-sector. For dual 13/16-sector images (eg. Taipan.woz) then prefer 16-sector firmware.

So if you aren't using a woz image, then currently you are out of luck, and will have to use the START13/BOOT13 workflow instead.

**Comment by skiphipdata** [comment]
Just to clarify, I'm not asking for auto-detection of 13-sector disk images at all. All I'm asking for is the ability to use a command line switch to manually activate the 13-sector Disk II ROM in place of the 16-sector ROM. Or, a switch that allows for the use of a custom ROM image contained in an external file for the 256-byte Disk II ROM (similar to the ability to use a custom ROM image for the main 12k or 16k system ROM.) The command line option could be -diskii13 or something similar, or -diskiirom <filename>.

**Comment by tomcw** [comment]
I've added a new command line '-s\<n\> diskii13' to put a diskii card into slot-n and force it to use the 13-sector firmware, eg:
```
applewin.exe -s6 diskii13
```

So you can mix Disk II cards with a 13-sector card in slot-5 and a 16-sector card in slot-6, eg:
```
applewin.exe -s5 diskii13 -s6 diskii
```

In this case, the slot-5 card is fixed with 13-sector firmware, but the slot-6 card will default to 16-sector firmware (unless a 13-sector WOZ image is inserted).

NB. there's currently no way to enable for this via the Configuration GUI, so you must use the command line for now.

**Comment by tomcw** [comment]
#### Developer note:
Think of the above "diskii13" support as a physical switch on the Disk II card to select between the 13 and 16 sector firmwares.

Originally I played with sub-classing `Disk2InterfaceCard` as `Disk2Interface13Card`, but then the static function `GetSnapshotCardName()` collided, so that needed a new `GetSnapshotCardName13()`. Also this introduced a new `SS_CARDTYPE CT_Disk2_13`... overall it was more code, and messier.

One pro was that the new `SS_CARDTYPE CT_Disk2_13` encapsulated this `force13SectorFirmware`, which implicitly saved it to the Registry. Currently this isn't done, and relies on the command line switch to enable it. When this is added to the Configuration GUI, then a new Registry entry will be needed.

**Comment by skiphipdata** [comment]
Thank you, that addresses the request very nicely. It also probably takes care of the issue where removing the Disk II card from Slot 6 with -s6 empty results in it being permanently removed, with editing the registry appearing to be the only way to put it back. There does not appear currently to be a -s6 diskii option like there is for Slot 5. That was going to be my next request anyway :)

**Comment by tomcw** [comment]
>It also probably takes care of the issue where removing the Disk II card from Slot 6...

Yes, now fixed that too!

**Comment by tomcw** [comment]
Released this in [AppleWin 1.30.13.0](https://github.com/AppleWin/AppleWin/releases/tag/v1.30.13.0), so closing this issue.
